WebView/app/src/main/java/com/android/samples/webviewdemo/MainActivity.kt

```diff
@@ -51,55 +54,55 @@ class MainActivity : AppCompatActivity() {
         }
     }
 
-    /** Instantiate the interface and set the context  */
-    class WebAppInterface(private val mContext: Context) {
-        /** Send a message from the web page  */
-        @JavascriptInterface
-        fun sendMessage(message: String) {
-            val sendIntent: Intent = Intent().apply {
-                action = Intent.ACTION_SEND
-                putExtra(Intent.EXTRA_TEXT, message)
-                type = "text/plain"
-            }
-            val shareIntent = Intent.createChooser(sendIntent, null)
-            startActivity(mContext, shareIntent, null)
+    private fun createJsObject(
+        webview: WebView,
+        mContext: Context,
+        jsObjName: String,
+        rules: Set<String>,
+        sendAndroidMessage: (message: String, mContext: Context) -> Unit
```

Comment:
Could we make this a simpler callback (only accepts a String) and remove the `mContext` arg from `createJsObject()`?

Also, a more conventional name for a callback method would be `onMessageReceived`.

Reply:
I think we need the context to start the activity in onMessageReceived on line 96

Reply:
You can probably skip the context by using `this`, since it should have access to the `Activity` context.

Reply:
Yup, that works, thanks :)